src/python/pants/backend/experimental/python/lockfile_metadata_test.py

```diff
@@ -15,8 +16,14 @@
 
 def test_metadata_round_trip() -> None:
     val = "help_i_am_trapped_inside_a_unit_test_string"
```

Comment:
Originally the point of the code was to show that the metadata (de)serialize functions are able to preserve arbitrary data, which I think is more important than showing that it can preserve an actual hash. 

On the other hand, `read_lockfile_metadata` is now calling `InterpreterConstraints`, which probably shows that that particular function has poor separation of concerns. Thinking about that now.

Reply:
I landed on a better organisation for `lockfile_metadata.py`, so that's good.